index.js

```diff
@@ -154,20 +158,30 @@ function applyAttributes(t, openingElement, componentName, sourceFileName, compo
   }
 }
 
-function processJSXElement(t, jsxElement, componentName, sourceFileName, componentAttributeName, elementAttributeName, sourceFileAttributeName) {
+function processJSXElement(annotateFragments, t, jsxElement, componentName, sourceFileName, componentAttributeName, elementAttributeName, sourceFileAttributeName) {
   if (!jsxElement) {
     return
   }
+  const openingElement = jsxElement.get('openingElement')
+
+  applyAttributes(t, openingElement, componentName, sourceFileName, componentAttributeName, elementAttributeName, sourceFileAttributeName)
 
-  applyAttributes(t, jsxElement.get('openingElement'), componentName, sourceFileName, componentAttributeName, elementAttributeName, sourceFileAttributeName)
   const children = jsxElement.get('children')
   if (children && children.length) {
-    // Children don't receive the data-component attribute so we pass null for componentName
-    children.forEach(element => processJSXElement(t, element, null, sourceFileName, componentAttributeName, elementAttributeName, sourceFileAttributeName))
+    let shouldSetComponentName = annotateFragments
+    for (let i=0; i < children.length; i += 1){
+      // Children don't receive the data-component attribute so we pass null for componentName unless it's the first child of a Fragment with a node and `annotateFragments` is true
+      if (shouldSetComponentName && children[i].get('openingElement') && children[i].get('openingElement').node) {
+        shouldSetComponentName = false
+        processJSXElement(annotateFragments, t, children[i], componentName, sourceFileName, componentAttributeName, elementAttributeName, sourceFileAttributeName, annotateFragments)
```

Comment:
this doesn't check whether the parent is an attribute before applying the component name. Is it possible that a child component of a non-fragment component will be decorated with a `data-component` attribute when it should be decorated with a `data-element` attribute?

Reply:
Those attributes aren't mutually exclusive so it's OK.